docs/ref/class-based-views/base.txt

```diff
@@ -70,11 +71,21 @@ MRO is an acronym for Method Resolution Order.
         attributes.
 
         When the view is called during the request/response cycle, the
-        :class:`~django.http.HttpRequest` is assigned to the view's ``request``
-        attribute. Any positional and/or keyword arguments :ref:`captured from
-        the URL pattern <how-django-processes-a-request>` are assigned to the
-        ``args`` and ``kwargs`` attributes, respectively. Then :meth:`dispatch`
-        is called.
+        :meth:`setup` method assigns the :class:`~django.http.HttpRequest` to
+        the view's ``request`` attribute, and any positional and/or keyword
+        arguments :ref:`captured from the URL pattern
+        <how-django-processes-a-request>` to the ``args`` and ``kwargs``
+        attributes, respectively. Then :meth:`dispatch` is called.
+
+    .. method:: setup(request, *args, **kwargs)
+
+        .. versionadded:: 2.2
+
+        Initializes view instance attributes: ``self.request``, ``self.args``,
+        and ``self.kwargs`` prior to :meth:`dispatch`.
+
+        Overriding this method allows mixins to setup instance attributes for
+        reuse in child classes. Child classes must call ``super()``.
```

Comment:
I preferred the shorter version we had earlier: 

```
Initializes view instance attributes: ``self.request``, ``self.args``
and ``self.kwargs`` prior to :meth:`dispatch`. Child classes overriding
this method must call ``super()``.
```

Reply:
I restored it from https://github.com/django/django/compare/53d05c3d166eb2f5e410b7a6c2c947a6480b1c50..c1022f47e051924288d2d5cfbdf7c7cfecd59a79#diff-2df22e647d63040c7db3eca921c888fdL87, I believe this change was part of @timgraham's cosmetic edits.

The longer version explains the purpose of this hook, that's why I have a slight preference for it.